cli/cmd/deploy/deploy.go

```diff
@@ -65,7 +69,25 @@ func DeployCmd(ch *cmdutil.Helper) *cobra.Command {
 		}
 	}
 
+	// the older `upload`` flag(which used tarballs) is now deprecated in favour of `managed` flag(which uses rill managed git repos)
+	// but we still keep it for backwards compatibility and use it for same behaviour as managed. Summarily:
+	// --managed and --upload are mutually exclusive and do the same thing i.e. create a project using rill managed git repo
+	// --archive is hidden and create a project using tarballs. It is for testing only.
+	// --github creates a project using user managed github repo
+	deployCmd.Flags().BoolVar(&upload, "managed", false, "Create project using rill managed repo")
 	deployCmd.Flags().BoolVar(&upload, "upload", false, "Create project using rill managed repo")
+	err := deployCmd.Flags().MarkDeprecated("upload", "Use --managed instead")
+	if err != nil {
+		panic(err)
+	}
+	deployCmd.MarkFlagsMutuallyExclusive("managed", "upload")
```

Comment:
I think we should just remove the `upload` flag. It's more okay to change CLI behavior than APIs because compatibility is not a problem.

One worry I had was for folks deploying from Gitlab, but I checked and our docs for Gitlab deploys don't use the `--upload` flag, so I don't think there's anything to be worried about here.